packages/fxa-content-server/app/scripts/lib/auth/crypto.ts

```diff
@@ -0,0 +1,179 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { hexToUint8, uint8ToHex, xor } from './utils';
+
+const encoder = () => new TextEncoder();
+const NAMESPACE = 'identity.mozilla.com/picl/v1/';
+
+type hexstring = string;
+
+export async function getCredentials(email: string, password: string) {
+  const passkey = await crypto.subtle.importKey(
+    'raw',
+    encoder().encode(password),
+    'PBKDF2',
+    false,
+    ['deriveBits']
+  );
+  const quickStretchedRaw = await crypto.subtle.deriveBits(
+    {
+      name: 'PBKDF2',
+      salt: encoder().encode(`${NAMESPACE}quickStretch:${email}`),
+      iterations: 1000,
+      hash: 'SHA-256',
+    },
+    passkey,
+    256
+  );
+  const quickStretchedKey = await crypto.subtle.importKey(
+    'raw',
+    quickStretchedRaw,
+    'HKDF',
+    false,
+    ['deriveBits']
+  );
+  const authPW = await crypto.subtle.deriveBits(
+    {
+      name: 'HKDF',
+      salt: new Uint8Array(0),
+      // @ts-ignore
```

Comment:
The builtin typescript definition for HKDF is wrong. We need to ignore the `info` field for now